docs/translations.md

```diff
@@ -0,0 +1,270 @@
+
+The list of values required **must** be defined in a separate file due to technical limitations imposed by React. Any module using 'use client' can only export components and no properties can be accessed from them. 
+
+To work around this limitation, the translations key list is defined in */src/app/translations/\**
+
+/src/app/translations/searchFilter.ts
+```ts
+export const searchFilterTranslationKeys = [
+    "price range",
+    "is available",
+    "search",
+    "filter",
+    "apply",
+] as const;
+
+export type SearchFilterTranslationKeys = typeof searchFilterTranslationKeys[number];
+```
+
+### Bundling Translations for Client Side
+
+The dictionaries are primarily located on the server side, and a client-side component may require many translation values. The best way to pass the values is in a single translation property as the alternative would be to pass as individual props. 
+
+To get a bundle of translation values we can make use of the *useTranslationValues* hook.
+
+*src/core/server/useTranslationValues.ts*
+```ts
+export const useTranslationValues = <Keys extends string>(
+  translationKeys: readonly Keys[],
+  namespace?: string,
+) => { 
+    //...
+}
+```
+Arguments:
+- *translationKeys*: an array of key values to get from the dictionary 
+- *namespace*: a dot-separated path that describes the location of the keys. This path is relative to the [dictionary schema](#dictionary-schema) that was defined.
+
+Example usage:
+```ts
+ const searchFilterTranslations = useTranslationValues(
+    searchFilterTranslationKeys,
+    "component"
+  );
+```
+
+If a translation value is not found in the locale dictionary then the output will be in the form of "(path.to.translation.value)". 
+
+## Navigation
+
+The *locale* is defined by the request route along with the *channel* of the storefront. The path has the pattern /[locale]/[channel]/\<path>. When navigating, it is required for the client-side to be aware of what the locale and channel values are so they may persist from page-request to page-request. It is possible to parse these values from *window.location*, but this logic is better placed in a dedicated module that can work automatically with the links. 
+
+
+### App Provider
+The **AppProvider** has the role of holding the *locale* and *channel* information and making it available to all client-side components that need it. Its definition is located in */src/core/client/useApp.tsx*
+
+### Link Component
+
+A custom **Link** component has been defined in */src/[locale]/(components)/link.tsx* that makes use of the AppProvider context information to provide the correct locale and channel information for navigation. It uses the NextJS \<Link> component as a base and adds the option to override the *locale* or *channel*. If no locale or channel props are provided, then it will keep the current values.
+
+```tsx
+<Link locale="fr-fr" channel="default-channel">
```

Comment:
I wonder if we should name this `CommerceLink` or `SaleorLink` or something like that? 

I kept accidentally importing the wrong link type when trying to use this, and it's hard to tell just by looking at it when the URL is external, or has no defined `locale` or `channel` prop.

Like maybe `<Link>` is just the regular link and `<CommerceLink>` (or whatever we want to call it) is the the one you know is specific to the starter kit. 

That would also simplify the implementation of the link component itself, which has to check whether the URL is local or not currently: https://github.com/limlabs/saleor-storefront-starter/blob/main/src/app/%5Blocale%5D/(components)/link.tsx#L26